lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/FeatureService.java

```diff
@@ -265,19 +270,23 @@ public List<Feature> getFeatureReplacements(String featureName, DOMNode featureM
         for (String nextFeatureName : featureNamesLowerCase) {
             if (existingFeatures.contains(nextFeatureName)) {
                 // collect feature name minus version number to know which other features to exclude
-                String featureNameMinusVersion = nextFeatureName.substring(0, nextFeatureName.lastIndexOf("-") + 1);
+                String featureNameMinusVersion = nextFeatureName.contains("-") ?
+                        nextFeatureName.substring(0, nextFeatureName.lastIndexOf("-") + 1) : nextFeatureName;
                 featuresWithoutVersionsToExclude.add(featureNameMinusVersion);
             }
         }
 
         List<Feature> replacementFeatures = new ArrayList<Feature>();
         String featureNameLowerCase = featureName.toLowerCase();
 
-        for (int i=0; i < featureNamesLowerCase.size(); i++) {
+        for (int i = 0; i < featureNamesLowerCase.size(); i++) {
             String nextFeatureName = featureNamesLowerCase.get(i);
-            if (nextFeatureName.contains(featureNameLowerCase) && (!nextFeatureName.contains("-") || 
-                !featuresWithoutVersionsToExclude.contains(nextFeatureName.substring(0, nextFeatureName.lastIndexOf("-") + 1)))) {
+            if (nextFeatureName.contains(featureNameLowerCase)) {
+                String comparingFeatureName = nextFeatureName.contains("-") ?
+                        nextFeatureName.substring(0, nextFeatureName.lastIndexOf("-") + 1) : nextFeatureName+"-";
```

Comment:
Yes its needed in both places.

Per my logic, we are just adding a trailing "-" , it will ignore both versioned and versionless features of same type

With this logic, 
1. if user already has jaxrs, all features starts with jaxrs- will be removed from completion
2. if user already has jaxrs-2.1, all features starts with jaxrs- will be removed from completion